qutebrowser/misc/sessions.py

```diff
@@ -142,6 +142,11 @@ def _save_tab(self, tab, active):
                 'url': bytes(item.url().toEncoded()).decode('ascii'),
                 'title': item.title(),
             }
+            if not item_data['title'] and idx == 0: #Most recent first
+                item_data['title'] = tab.title()
```

Comment:
The user won't always have the first item open - this probably should be in `if history.currentItemIndex() == idx:` below.